applications/nrf5340_audio/src/bluetooth/ble_audio_services.c

```diff
@@ -272,14 +415,66 @@ int ble_vcs_client_init(void)
 {
 	static struct bt_vcs_cb vcs_client_callback;
 
-	vcs_client_callback.discover = vcs_dis_cb_handler;
+	vcs_client_callback.discover = vcs_discover_cb_handler;
 	vcs_client_callback.state = vcs_state_cb_handler;
 	vcs_client_callback.flags = vcs_flags_cb_handler;
 	return bt_vcs_client_cb_register(&vcs_client_callback);
 }
 #endif /*CONFIG_BT_VCS_CLIENT*/
 #endif /*(CONFIG_AUDIO_DEV == GATEWAY)*/
 
+int ble_mcs_discover(struct bt_conn *conn)
+{
+#if CONFIG_BT_MCC
+	return bt_mcc_discover_mcs(conn, true);
+#else
+	return 0;
+#endif
+}
+
+int ble_mcs_state_update(struct bt_conn *conn)
+{
+#if CONFIG_BT_MCC
+	return bt_mcc_read_media_state(conn);
+#else
+	return 0;
+#endif
+}
+
+int ble_mcs_play_pause(struct bt_conn *conn)
+{
+	int ret;
+	struct mpl_cmd cmd;
+
+#if CONFIG_BT_MCS
+	media_proxy_ctrl_get_media_state(local_player);
+#endif
+
+	if (media_player_state == BT_MCS_MEDIA_STATE_PLAYING) {
```

Comment:
To me this looks like a race condition/race hazard.
What happens if mcs_media_state_cb() or mcc_read_media_state_cb() is called too late?

Reply:
The only potential problem here is that the operation will be done on an old state. So the user might have to press pause twice if the stream is not paused right away. Since we have the same button for both operation I don't see this as a huge issue. Also, I don't see how we would guard against this.

Reply:
As discussed, remove media_proxy_ctrl_get_media_state()